pkg/util/converter/prom.go

```diff
@@ -0,0 +1,337 @@
+			default:
+				iter.Skip()
+				rsp = &backend.DataResponse{
+					Error: fmt.Errorf("unknown result type: %s", resultType),
+				}
+			}
+
+		case "stats":
+			v := iter.Read()
+			fmt.Printf("[data] TODO, support stats: %v\n", v)
+
+		default:
+			v := iter.Read()
+			fmt.Printf("[data] TODO, support key: %s / %v\n", l1Field, v)
+		}
+	}
+
+	fmt.Printf("result: %s\n", resultType)
+	return rsp
+}
+
+// will always return strings for now
+func readArrayData(iter *jsoniter.Iterator) *backend.DataResponse {
+	lookup := make(map[string]*data.Field)
+
+	rsp := &backend.DataResponse{}
+	stringField := data.NewFieldFromFieldType(data.FieldTypeString, 0)
+	stringField.Name = "Value"
+	for iter.ReadArray() {
+		switch iter.WhatIsNext() {
+		case jsoniter.StringValue:
+			stringField.Append(iter.ReadString())
+
+		// Either label or exemplars
+		case jsoniter.ObjectValue:
+			f, pairs := readLabelsOrExemplars(iter)
+			if f != nil {
+				rsp.Frames = append(rsp.Frames, f)
+			} else if pairs != nil {
+				max := 0
+				for _, pair := range pairs {
+					k := pair[0]
+					v := pair[1]
+					f, ok := lookup[k]
+					if !ok {
+						f = data.NewFieldFromFieldType(data.FieldTypeString, 0)
+						f.Name = k
+						lookup[k] = f
+
+						if len(rsp.Frames) == 0 {
+							rsp.Frames = append(rsp.Frames, data.NewFrame(""))
+						}
+						rsp.Frames[0].Fields = append(rsp.Frames[0].Fields, f)
+					}
+					f.Append(fmt.Sprintf("%v", v))
+					if f.Len() > max {
+						max = f.Len()
+					}
+				}
+
+				for _, f := range lookup {
+					if f.Len() != max {
+						f.Append("") // no matching label
+					}
+				}
+			}
+
+		default:
+			{
+				ext := iter.ReadAny()
+				v := fmt.Sprintf("%v", ext)
+				stringField.Append(v)
+			}
+		}
+	}
+
+	if rsp.Frames == nil {
+		rsp.Frames = data.Frames{data.NewFrame("", stringField)}
+	}
+
+	return rsp
+}
+
+// For consistent ordering read values to an aray not a map
+func readLabelsAsPairs(iter *jsoniter.Iterator) [][2]string {
+	pairs := make([][2]string, 0, 10)
+	for k := iter.ReadObject(); k != ""; k = iter.ReadObject() {
+		pairs = append(pairs, [2]string{k, iter.ReadString()})
+	}
+	return pairs
+}
+
+func readLabelsOrExemplars(iter *jsoniter.Iterator) (*data.Frame, [][2]string) {
+	pairs := make([][2]string, 0, 10)
+	labels := data.Labels{}
+	var frame *data.Frame
+
+	for l1Field := iter.ReadObject(); l1Field != ""; l1Field = iter.ReadObject() {
+		switch l1Field {
+		case "seriesLabels":
+			iter.ReadVal(&labels)
+		case "exemplars":
+			lookup := make(map[string]*data.Field)
+			timeField := data.NewFieldFromFieldType(data.FieldTypeTime, 0)
+			valueField := data.NewFieldFromFieldType(data.FieldTypeFloat64, 0)
+			valueField.Name = labels["__name__"]
+			delete(labels, "__name__")
+			valueField.Labels = labels
+			frame = data.NewFrame("", timeField, valueField)
+			for iter.ReadArray() {
+				for l2Field := iter.ReadObject(); l2Field != ""; l2Field = iter.ReadObject() {
+					switch l2Field {
```

Comment:
any reason why this `switch` has no `default` (maybe with return-error)?

Reply:
added a return with notices